src/user-search/client.ts

```diff
@@ -0,0 +1,294 @@
+import {Client, ExtendedAccount} from 'dsteem'
+import * as _ from 'lodash'
+import * as moment from 'moment'
+const nodecache = require( 'node-cache' )
+import * as config from 'config'
+import { Dictionary } from 'lodash'
+import {logger} from '../logger'
+import {UserAccount, UserAccountJSON, UserContext, UserContextJSON} from './user'
+import {loadAccountNames} from './indexes'
+
+const STEEMD_API_URL = 'https://api.steemit.com'
+const CACHE_CLIENT_TTL = 600 // config.get('cacheClient')['ttl'] FIXME
+const CACHE_CLIENT_CHECK_INTERVAL = 60 // config.get('cacheClient')['interval'] FIXME
+enum FollowType {undefined, blog, ignore}
+
+export interface FollowCountReturn {
+    account: string
+    follower_count: number
+    following_count: number
+}
+
+export interface FollowReturn {
+    follower: string
+    following: string
+    what: string[]
+}
+
+export type TransferTargetCounts = Dictionary<number>
+
+interface AccountHistoryItem {
+    block: number
+    op: [string, object]
+    op_in_trx: number
+    timestamp: string
+    trx_id: string
+    trx_in_block: number
+    virtual_op: number
+}
+
+interface AccountHistoryPair {
+    [index: number]: [number, AccountHistoryItem]
+}
+
+export class CachingClient {
+    private readonly pageSize: number = 1000
+
+    constructor(public readonly cache?: any,
+                public readonly cacheOptions = {stdTTL: CACHE_CLIENT_TTL, checkperiod: CACHE_CLIENT_CHECK_INTERVAL},
+                private readonly client?: any,
+                public readonly address = STEEMD_API_URL) {
+        if (cache === undefined) {
+            this.cache = new nodecache(cacheOptions)
+        } else {
+            this.cache = cache
+        }
+        if (client === undefined) {
+            this.client = new Client(address)
```

Comment:
what is `Client`?

Reply:
ah nevermind my bad

Reply:
client is either the  `dsteem` client or a `sinon.fake` that the constructor accepts. I left it untyped here to help with my tests